src/HTTPure/ResponseHeaders.purs

```diff
@@ -0,0 +1,86 @@
+module HTTPure.ResponseHeaders
+  ( ResponseHeaders(..)
+  , empty
+  , headers
+  , headers'
+  , header
+  , header'
+  , write
+  ) where
+
+import Prelude
+
+import Data.Array.NonEmpty (NonEmptyArray, toArray)
+import Data.Array.NonEmpty as NonEmptyArray
+import Data.Foldable (foldl)
+import Data.FoldableWithIndex (foldMapWithIndex)
+import Data.Map (Map, singleton, union, insert)
+import Data.Map (empty) as Map
+import Data.Newtype (class Newtype, unwrap)
+import Data.String.CaseInsensitive (CaseInsensitiveString(CaseInsensitiveString))
+import Data.Traversable (traverse)
+import Data.TraversableWithIndex (traverseWithIndex)
+import Data.Tuple (Tuple(Tuple))
+import Effect (Effect)
+import HTTPure.Lookup (class Lookup, (!!))
+import Node.HTTP (Response, setHeader)
+
+-- | The `ResponseHeaders` type is just sugar for a `Map` of `Strings`
+-- | that represents the set of headers in an HTTP request or response.
+newtype ResponseHeaders = ResponseHeaders (Map CaseInsensitiveString (NonEmptyArray String))
+
+derive instance newtypeResponseHeaders :: Newtype ResponseHeaders _
+
+-- | Given a string, return a `Maybe` containing the value of the matching
+-- | header, if there is any.
+instance lookupResponseHeaders :: Lookup ResponseHeaders String (NonEmptyArray String) where
+  lookup (ResponseHeaders responseHeaders) key = responseHeaders !! key
+
+-- | Allow a `ResponseHeaders` to be represented as a string. This string
+-- | is formatted in HTTP headers format.
+instance showResponseHeaders :: Show ResponseHeaders where
+  show (ResponseHeaders responseHeaders) = foldMapWithIndex showField responseHeaders <> "\n"
+    where
+    showField key value = unwrap key <> ": " <> NonEmptyArray.intercalate "," value <> "\n"
```

Comment:
This `show` implementation is probably going to break on `Set-Cookie` headers with `expires` dates, no? E.g.:
```
Set-Cookie: id=a3fWa; Expires=Wed, 21 Oct 2015 07:28:00 GMT; Secure; HttpOnly
```

Reply:
fyi: I implemented this separation recently in my httpure fork: https://github.com/sigma-andex/purescript-httpurple/pull/12/files
I think what you are missing here is that node has `setHeaders` function which handles multiple response headers.

Reply:
here is the corresponding line: https://github.com/sigma-andex/purescript-httpurple/pull/12/files#diff-34348f738f2161e53492eb030b6dc6e776b34389619b3a041180a7be3d7864d8R97

Reply:
Oh, my bad, I completely missed out on the `HTTPure.ResponseHeaders.write` function. Yeah, it looks like we should be using `Node.HTTP.setHeaders` there instead of `Node.HTTP.setHeader`. Thanks for the tip @sigma-andex!

Reply:
As I mentioned for `RequestHeaders`, I don't think this `show` instance  is right. It should produce something more like:

```purs
(ResponseHeaders (fromFoldable [Tuple (CaseInsensitiveString key) (cons' value1 [value2])))
```

Reply:
And might also be derivable with a `Generic` instance.